safe_autonomy_simulation/sims/inspection/sun.py

```diff
@@ -54,8 +54,7 @@ def __eq__(self, other):
         return False
 
     def build_initial_state(self) -> np.ndarray:
-        state = super().build_initial_state()
-        state = np.concatenate([state, np.array([self._initial_theta])])
+        state = np.array([self._initial_theta])
         return state
```

Comment:
The sun entity should still maintain the full internal state expected of a PhysicalEntity. Otherwise, the properties returned by the sun entity other than ones explicitly implemented in the Sun() class will return incorrect values compared to what a Physical Entity is expected to present. The original implementation for this method should be correct and kept.

Reply:
The failure of the Sun() entity to update theta seems to be due to the state setter setting the wrong index in the internal state array. It should set `self._state[-1] = state[0]`. The current implementation is setting the x position index of the internal state vector to theta instead.

Reply:
We could avoid magic numbers with a class enum that indexes the internal state but that's outside of this scope.

Reply:
I've pushed a commit which implements this update and confirmed that `Sun().step(num_steps)` does indeed increment `Sun().theta` by `num_steps * mean_motion`

Reply:
This should be fine.  The original change was done because the Sun hides the position from `_state` due to the overriding of the `position` property.  However, it does make sense to try to keep the `_state` consistent with the base class even if values are hidden.